src/main.cpp

```diff
@@ -4320,52 +4319,69 @@ bool ServicesIncludesNN(CNode* pNode)
     return (Contains(pNode->strSubVer,"1999")) ? true : false;
 }
 
-bool VerifySuperblock(std::string superblock, int nHeight)
+bool VerifySuperblock(const std::string& superblock, const CBlockIndex* parent)
 {
-        bool bPassed = false;
-        double out_avg = 0;
-        double out_beacon_count=0;
-        double out_participant_count=0;
-        double avg_mag = 0;
-        if (superblock.length() > 20)
-        {
-            avg_mag = GetSuperblockAvgMag(superblock,out_beacon_count,out_participant_count,out_avg,false,nHeight);
-            bPassed=true;
-            if (!IsResearchAgeEnabled(nHeight))
-            {
-                return (avg_mag < 10 ? false : true);
-            }
-            // New rules added here:
-            if (out_avg < 10 && fTestNet)  bPassed = false;
-            if (out_avg < 70 && !fTestNet) bPassed = false;
-            if (avg_mag < 10 && !fTestNet) bPassed = false;
-			// Verify distinct project count matches whitelist
-        }
-        if (fDebug3 && !bPassed)
-        {
-            if (fDebug) printf(" Verification of Superblock Failed ");
-            //if (fDebug3) printf("\r\n Verification of Superblock Failed outavg: %f, avg_mag %f, Height %f, Out_Beacon_count %f, Out_participant_count %f, block %s", (double)out_avg,(double)avg_mag,(double)nHeight,(double)out_beacon_count,(double)out_participant_count,superblock.c_str());
-        }
-        return bPassed;
+    // Pre-condition checks.
+    if(!parent)
+    {
+        printf("Invalid block passed to VerifySuperblock");
+        return false;
+    }
+
+    if(superblock.length() <= 20)
+    {
+        printf("Invalid superblock passed to VerifySuperblock");
+        return false;
+    }
+
+    // Validate superblock contents.
+    bool bPassed = true;
+    double out_avg = 0;
+    double out_beacon_count=0;
+    double out_participant_count=0;
+    double avg_mag = GetSuperblockAvgMag(superblock,out_beacon_count,out_participant_count,out_avg,false, parent->nHeight);
+
+    // New rules added here:
+    if (out_avg < 10 && fTestNet)  bPassed = false;
+    if (out_avg < 70 && !fTestNet) bPassed = false;
+
+    // Before block version- and stake engine 8 there used to be a requirement
+    // that the average magnitude across all researchers must be at least 10.
+    // This is not necessary but cannot be changed until a mandatory is released.
+    //
+    // TODO: Remove this after V8 has kicked in?
+    if(!fTestNet && parent->nVersion < 8 && avg_mag < 10)
```

Comment:
Why version condition is different on testnet and prod?
New version is enabled separately for prod and test, so that hides the difference.